### DISCUSSION TITLE
Multi-line edit support

### DISCUSSION THREAD

**Discussion by abcd-ca** [zed-industries/zed#6524] [Feature Requests]
Hi, I'm looking to find the keymap to add multiple cursors so that I can do a multi-line edit. In my other IDE I have a keycombo with the down arrow key that adds a cursor to the next line for as many lines as I need then I just start moving the cursor around and they all move and edit together on multiple lines like this:

![out](https://user-images.githubusercontent.com/1296103/224817542-d7e8adf4-0bbe-485f-81d1-2f99b5f92024.gif)

Does it exist in Zed? Thank you.

**Comment by mikayla-maki** [comment]
Yes! Use alt-cmd-[up/down] to copy your cursor up or down :) you can also `alt-click` to place a multi-cursor in a specific spot.

**Reply by tcurdt** [reply]
While alt-cmd-up/down works for me, alt-click does not (on macOS 15).
Any pointers what I might be missing?

**Reply by bahrom04** [reply]
hold shift+option and click. it works on macOs

**Reply by tcurdt** [reply]
A shame it's shift+option-click instead of the standard option-click.
But at least it works. Thanks!

**Reply by Stormcasta** [reply]
Can the shift+option-click be configured to remap to option-click? Also, you can't click and drag to select multiple lines to edit, you have to individually click each edit point.

**Reply by injust** [reply]
opt-click is working for me on macOS.

shift-opt-click creates a rectangular selection, which makes sense when you think about how the shift modifier should affect opt-click.

**Reply by Stormcasta** [reply]
ah, I see. The way it's implemented in Zed is more sensible, in VSCode you get the shift-opt-click behaviour if you click and drag but the opt-click behaviour if you don't drag.

**Reply by tcurdt** [reply]
@injust well, it's multi-select but not a multi-line select. 

Given how often I would use one or the other ...and what has been basically the default for more than a decade I would appreciate if:
1. shit-option-click and option-click were reversed
2. I can re-assign/switch the two by configuration

**Comment by abcd-ca** [comment]
Yasss! Thanks, that was fast :)

**Comment by abcd-ca** [comment]
I don't see it in the default bindings https://zed.dev/ref/default.json

**Comment by erf** [comment]
The [vis](https://github.com/martanne/vis) editor combines multiple cursors with sam's regular expressions. I really like this feature where you are able to do a regex search for selections, a range with an anchor, from which you are able to edit text.

**Comment by samgaw** [comment]
This was one of the first things I looked for when Cmd+click didn't work and couldn't find it in the docs or default bindings. Can this be exposed for remapping?

**Reply by benswift** [reply]
did we ever get an answer to whether this can be re-mapped?

**Reply by benswift** [reply]
Oh, I found it - the command is `editor::AddSelectionBelow`. So yep, you can re-bind it just like any other command.

**Reply by atuttle** [reply]
Thank you so much for sharing this. It's still missing from the documentation!

**Reply by lukasgolino** [reply]
Man thank you, I spent ages trying to find simply the name of this damn keybind!
